src/main/scala/mesosphere/marathon/core/group/GroupManager.scala

```diff
@@ -130,14 +130,21 @@ trait GroupManager {
     fn: RootGroup => RootGroup,
     version: Timestamp = Group.defaultVersion,
     force: Boolean = false,
+    toKill: Map[PathId, Seq[Instance]] = Map.empty): Future[DeploymentPlan] = updateRootAsync(id, (g: RootGroup) => Future.successful(fn(g)), version, force, toKill)
+
+  final def updateRootAsync(
+    id: PathId,
+    fn: RootGroup => Future[RootGroup],
+    version: Timestamp = Group.defaultVersion,
+    force: Boolean = false,
     toKill: Map[PathId, Seq[Instance]] = Map.empty): Future[DeploymentPlan] = {
-    updateRootEither[Nothing](id, fn.andThen(Right(_)), version, force, toKill)
+    updateRootEither[Nothing](id, fn(_).map(Right(_))(ExecutionContexts.callerThread), version, force, toKill)
```

Comment:
Do we want this? Was this a bug before?

Reply:
I needed to do that because it now operates on future ... it is not just a method composition as it was before